### PR TITLE
Allow for differing clustersets/additional trees in homology relationships

### PR DESCRIPTION
**Pull request by CristiGuijarro**
### CheckHomology

:bangbang: | This needs to be merged to handover compara vertebrates
:---: | :---

This compara DC checks for redundancy in homology relationships - duplicate relationships for a pair of `gene_members`.
But it does not take into account the possibility that a homology relationship can be identical and inferred from two different trees.
This flexibility has been checked on the staging sites and in the compara API, and causes no issues.

* Non default clustersets with overlapping mlsss in
  in the strains/breeds and default can cause `gene_members`
  to have duplicated relationships in different trees

* These "redundant" or duplicate relationships are actually
  a sign of consistency between trees

* Whilst overlapping mlsss should be avoided in the future,
  the data is not at fault, not technically redundant and
  should be allowed

* Check now groups by gene_tree_root_id as well as
  gene_members

### CODE REVIEW

File: lib/Bio/EnsEMBL/DataCheck/Checks/CheckHomology.pm
Comment (marcoooo) [inline lib/Bio/EnsEMBL/DataCheck/Checks/CheckHomology.pm:56]: imho, since `CROSS JOIN` is filtered on `homology_id` it will produce the exact same result as an inner join, I am not sure if that has an impact on performance, (depend on how mysql optimize the resulting data access) but CROSS JOIN is generally a "I want to kill my server" kind of thing.

Comment (CristiGuijarro) [reply lib/Bio/EnsEMBL/DataCheck/Checks/CheckHomology.pm:56]: This is a horrible "legacy" style query that was simply taken from the java healthcheck version, and now altered. I honestly strongly dislike the entire query, but have been unable to improve upon it. We do need to get all the possible lines in the table because of the nature of the way the relationships are defined in the `homology` and `homology_member` tables - really the best we can hope for is an improved schema design/model to store these homology relationships :crossed_fingers:

Comment (marcoooo) [reply lib/Bio/EnsEMBL/DataCheck/Checks/CheckHomology.pm:56]: As I said the CROSS being filtered on `homology_id` the result is NOT all the possible lines though, only the matched ones. I was just concerned about MySQL doing the CROSS `then` filtering, or automatically move to a more simple `INNER`.

Comment (marcoooo) [reply lib/Bio/EnsEMBL/DataCheck/Checks/CheckHomology.pm:56]: I'll wait for @twalsh-ebi approval before merging anyway

Comment (CristiGuijarro) [reply lib/Bio/EnsEMBL/DataCheck/Checks/CheckHomology.pm:56]: @marcoooo - Thank you and I think what you say makes sense. I will write a ticket for improving the query on our side based on what you have said for another release, at the moment I don't think we have the time to test another change

Comment (marcoooo) [reply lib/Bio/EnsEMBL/DataCheck/Checks/CheckHomology.pm:56]: NP.

**Review by marcoooo — APPROVED** [review]

**Pull request merged**
